src/vos/vos_obj_cache.c

```diff
@@ -214,7 +214,7 @@ vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
 	*obj_p = NULL;
 
 	if (cont->vc_pool->vp_dying)
-		return -DER_SHUTDOWN; /* TODO: use a more targeted errno */
```

Comment:
What's the benefit of changing the return value from "-DER_SHUTDOWN" to "-DER_POOL_KILLED" ? From the vos_obj_hold() caller's perspective, it may more care about whether the pool service is still available or not, as for whether the pool will be destroyed or not after stopping the pool service, that seems not important for the caller, right?

Reply:
I guess it's more that -DER_SHUTDOWN means service is shutting down so maybe a little overloaded here whereas -DER_POOL_KILLED is more specifically, the pool is getting destroyed.  I'm somewhat indifferent on this.   The patch is mostly about moving the DAOS errors to DAOS.   This all may be moot if we move cart back to daos.

Reply:
I understand that the pool is being destroyed here, but do we (or the caller) need to distinguish the case of the pool being destroyed from normal pool service shutting down?

Reply:
I reverted it.   I guess the question now is if we are going to move cart to daos, we probably don't need this at all.